nupic/research/frameworks/continuous_learning/multihead/subdataset.py

```diff
@@ -0,0 +1,191 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+Dataset functionality for MNIST and Google Speech Commands in a continual learning
+framework.
+"""
+
+import os
+
+import numpy as np
+from torch.utils.data import Dataset
+from torchvision import transforms
+from torchvision.datasets import MNIST
+
+
+def get_datasets(dataset_name, scenario, download=False):
+    """
+    Return the train and test datasets (in parallel lists) for each task in the
+    continuous learning setup
+    :param dataset_name: name of the dataset to retrieve
+    :type dataset_name: str
+    :param scenario: continuous learning setup, one of {'task', 'domain', 'class'}
+    :type scenario: str
+    :param download: whether or not to download the dataset
+    :type download: bool
+    :return: tuple of (list of pytorch dataset, list of pytorch dataset)
+    """
+    # specify image-transformations to be applied
+    dataset_transform = transforms.Compose([
+        transforms.ToTensor(),
+    ])
+
+    if dataset_name == "splitGSC":
+        dataset_class = GSC
+    elif dataset_name == "splitMNIST":
+        dataset_class = MNIST
+
+    full_train_set = dataset_class("~/nta/continual-learning/datasets/mnist/",
```

Comment:
I suggest to make the path to the dataset relative (possibly a command line argument in `train_model` or an environment variable), it will make it easier for others to run.

Reply:
I'll just use the root directory.